### DISCUSSION TITLE
start_standalone.sh  run failed with connecting to minio in macOS

### DISCUSSION THREAD

**Discussion by cherrylzhao** [milvus-io/milvus#38447] [Q&A and General discussion]
hi, everyone

I'm trying to debug milvus v2.4.14 in macOS 13.6.4, compile process is ok, but failing in run start_standalone.sh, error in /tmp/standalone.log is here

`
2024/12/13 16:49:19.770 +08:00] [ERROR] [querynode/service.go:144] ["QueryNode init error: "] [error="[UnknownError] precheck chunk manager client failed, error:Error in ListObjects[errcode:-1, exception:, errmessage:curlCode: 1, Unsupported protocol, params:params, bucket=a-bucket, prefix=justforconnectioncheck], configuration:[address=localhost:9000, bucket_name=a-bucket, root_path=files, storage_type=remote, cloud_provider=aws, iam_endpoint=, log_level=fatal, region=, useSSL=false, sslCACert=19, useIAM=false, useVirtualHost=false, requestTimeoutMs=10000]"] [errorVerbose="[UnknownError] precheck chunk manager client failed, error:Error in ListObjects[errcode:-1, exception:, errmessage:curlCode: 1, Unsupported protocol, params:params, bucket=a-bucket, prefix=justforconnectioncheck], configuration:[address=localhost:9000, bucket_name=a-bucket, root_path=files, storage_type=remote, cloud_provider=aws, iam_endpoint=, log_level=fatal, region=, useSSL=false, sslCACert=19, useIAM=false, useVirtualHost=false, requestTimeoutMs=10000]\n(1) attached stack trace\n  -- stack trace:\n  | github.com/milvus-io/milvus/internal/util/initcore.HandleCStatus
\n  | \t/Users/zhaojun/CLionProjects/milvus/internal/util/initcore/init_core.go:218\n  | github.com/milvus-io/milvus/internal/util/initcore.InitRemoteChunkManager\n  | \t/Users/zhaojun/CLionProjects/milvus/internal/util/initcore/init_core.go:171\n  | github.com/milvus-io/milvus/internal/querynodev2.(*QueryNode).InitSegcore\n  | \t/Users/zhaojun/CLionProjects/milvus/internal/querynodev2/server.go:244\n  | github.com/milvus-io/milvus/internal/querynodev2.(*QueryNode).Init.func1\n  | \t/Users/zhaojun/CLionProjects/milvus/internal/querynodev2/server.go:357\n  | sync.(*Once).doSlow\n  | \t/usr/local/go/src/sync/once.go:76\n  | sync.(*Once).Do\n  | \t/usr/local/go/src/sync/once.go:67\n  | github.com/milvus-io/milvus/internal/querynodev2.(*QueryNode).Init\n  | \t/Users/zhaojun/CLionProjects/milvus/internal/querynodev2/server.go:276\n  | github.com/milvus-io/milvus/internal/distributed/querynode.(*Server).init\n  | \t/Users/zhaojun/CLionProjects/milvus/internal/distributed/querynode/service.go:143\n  | github.com/milvus-io/milvus/internal/distributed/querynode.(*Server).Run\n  | \t/Users/zhaojun/CLionProjects/milvus/internal/distributed/querynode/service.go:222\n  | github.com/milvus-io/milvus/cmd/components.(*QueryNode).Run\n  | \t/Users/zhaojun/CLionProjects/milvus/cmd/components/query_node.go:59\n  | github.com/milvus-io/milvus/cmd/roles.runComponent[...].func1\n  | \t/Users/zhaojun/CLionProjects/milvus/cmd/roles/roles.go:126\n  | runtime.goexit\n  | \t/usr/local/go/src/runtime/asm_amd64.s:1700\nWraps: (2) [UnknownError] precheck chunk manager client failed, error:Error in ListObjects[errcode:-1, exception:, errmessage:curlCode: 1, Unsupported protocol, params:params, bucket=a-bucket, prefix=justforconnectioncheck], configuration:[address=localhost:9000, bucket_name=a-bucket, root_path=files, storage_type=remote, cloud_provider=aws, iam_endpoint=, log_level=fatal, region=, useSSL=false, sslCACert=19, useIAM=false, useVirtualHost=false, requestTimeoutMs=10000]\nError types: (1) *withstack.withStack (2) *errutil.leafError"] [stack="github.com/milvus-io/milvus/internal/distributed/querynode.(*Server).init\n\t/Users/zhaojun/CLionProjects/milvus/internal/distributed/querynode/service.go:144\ngithub.com/milvus-io/milvus/internal/distributed/querynode.(*Server).Run\n\t/Users/zhaojun/CLionProjects/milvus/internal/distributed/querynode/service.go:222\ngithub.com/milvus-io/milvus/cmd/components.(*QueryNode).Run\n\t/Users/zhaojun/CLionProjects/milvus/cmd/components/query_node.go:59\ngithub.com/milvus-io/milvus/cmd/roles.runComponent[...].func1\n\t/Users/zhaojun/CLionProjects/milvus/cmd/roles/roles.go:126"]
`

I'm sure minio service is ok in docker-compose

```
(base) zhaojun@mac ~/C/m/d/d/dev> docker-compose ps                                                                                                                  v2.4.14!?
     Name                    Command                  State                                                           Ports                                                    
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
dev_azurite_1     docker-entrypoint.sh azuri ...   Up             0.0.0.0:10000->10000/tcp, 10001/tcp, 10002/tcp                                                               
dev_etcd_1        etcd -listen-peer-urls=htt ...   Up             0.0.0.0:2379->2379/tcp, 0.0.0.0:2380->2380/tcp, 0.0.0.0:4001->4001/tcp                                       
dev_jaeger_1      /go/bin/all-in-one-linux         Up             14250/tcp, 0.0.0.0:14268->14268/tcp, 0.0.0.0:16686->16686/tcp, 0.0.0.0:4317->4317/tcp,                       
                                                                  0.0.0.0:4318->4318/tcp, 0.0.0.0:6831->6831/udp, 9411/tcp                                                     
dev_kafka_1       /opt/bitnami/scripts/kafka ...   Up             0.0.0.0:9092->9092/tcp                                                                                       
dev_minio_1       /usr/bin/docker-entrypoint ...   Up (healthy)   0.0.0.0:9000->9000/tcp, 0.0.0.0:9001->9001/tcp                                                               
dev_pulsar_1      /bin/bash -c                     Up             0.0.0.0:6650->6650/tcp, 0.0.0.0:18080->8080/tcp                                                              
                  bin/apply-co ...                                                                                                                                             
dev_zookeeper_1   /bin/sh -c /usr/sbin/sshd  ...   Up             0.0.0.0:2181->2181/tcp, 22/tcp, 2888/tcp, 3888/tcp `
```

here is mc test result
```
mc alias set myminio http://localhost:9000 minioadmin 
(base) zhaojun@mac ~/C/m/d/d/dev> mc ls myminio                                                                                                                      v2.4.14!?
[2024-11-11 17:16:40 CST]     0B a-bucket/
```

please give me some hint for that, thanks!

**Comment by yhmo** [comment]
Follow these steps to launch a local milvus:
(Note: the commands are running from milvus project root path.)

1. firstly ensure there is no milvus process running
```
$ ps -ef | grep milvus
$ source ./scripts/stop.sh
```

2. cmd to the project root path, set environment path
`$ source ./scripts/setenv.sh`

3. use a text tool to open the config/milvus.yaml, replace all the "/var/lib/milvus" to be "/tmp/milvus". This is to avoid permission deny to access the /var path.

4. start the third-party services, etcd/minio
```
$ cd deployments/docker/dev
$ docker-compose up -d
```


5. cmd to the project root path, start milvus
```
$ cd -
$ ./scripts/start_standalone.sh
```

After step 5, you will see a file /tmp/standalone.log generated, it is the log of milvus.

**Reply by cherrylzhao** [reply]
hi, yhmo, thanks for your reply!
I have follow above step, but same error still exist~

**Reply by yhmo** [reply]
I have no idea. Almost all milvus developers are using this way to run local milvus. Double-check your docker containers, clean all the containers and rerun the steps.

**Comment by xiaofan-luan** [comment]
@cherrylzhao 

did you test that inside the milvus docker itself it could work? 
you may need to make sure the minio is accessiable at milvus docker image.

check deployments/docker/standalone/docker-compose.yaml

this is what we use for deployment

**Reply by cherrylzhao** [reply]
I have tested, inside the milvus docker itself is ok.
so I wan to clone the milvus project, then try to make it again, 
how can I build milvus in debug? can I modify the mode of Makefile into 'Debug' ?

**Reply by cherrylzhao** [reply]
@xiaofan-luan
